network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py

```diff
@@ -35,9 +36,16 @@ class RecentBlogEntries(blocks.StructBlock):
         help_text="Optional divider below content block.",
     )
 
-    # TODO: add in validation so that if there are no tags or topic
-    #       filled in we don't allow the page to be saved, with a wagtail
-    #       error indication what's wrong.
+    def clean(self, value):
+        result = super().clean(value)
+
+        if result["tag_filter"] and result["topic_filter"]:
+            raise exceptions.ValidationError("Please provide either a Tag or a Topic, not both", code="invalid")
+
+        if not result["tag_filter"] and not result["topic_filter"]:
+            raise exceptions.ValidationError("Please provide a Tag or a Topic", code="required")
+
+        return result
```

Comment:
Really great work on the validation, @AdalbertoMoz! The clean method is well-organized and functions as expected. The use of the code parameter is a nice touch for future test scenarios. 👍

While reviewing this PR, I experimented with using StructBlockValidationError and found that it offers an added benefit. When a validation error occurs, it highlights the specific block containing the error, making it clearer for the user when they see the "This page could not be saved due to validation errors" message at the top.

Here are the differences I observed:


**ValidationError:**

![Screenshot 2024-06-10 at 12-25-58 Editing Donate help page Donate Help - Wagtail](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/50575ba8-8959-4c07-8107-81b1262538d5)

**StructBlockValidationError:** 

![Screenshot 2024-06-10 at 12-13-37 Editing Donate help page Donate Help - Wagtail](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/ffb026ef-eff5-4070-876d-290ff0d1cf7c)

If this approach sounds good to you, here's a suggested change:

```suggestion
        validation_errors = {}

        both_filters_error = ValidationError("Please provide either a Tag or a Topic, not both", code="invalid")
        no_filter_error = ValidationError("Please provide a Tag or a Topic", code="required")

        if value["tag_filter"] and value["topic_filter"]:
            validation_errors["tag_filter"] = both_filters_error
            validation_errors["topic_filter"] = both_filters_error

        if not value["tag_filter"] and not value["topic_filter"]:
            validation_errors["tag_filter"] = no_filter_error
            validation_errors["topic_filter"] = no_filter_error

        if validation_errors:
            raise StructBlockValidationError(validation_errors)

        return super().clean(value)
```

Feel free to reach out if you have any questions or would like any help. Great job again!

Reply:
Looks great Daniel! Thanks so much for the detailed review! 

I will push a commit with your suggested code and ensure that I import only the necessary functions, `ValidationError` and `StructBlockValidationError`, rather than importing the entire exceptions module.